### PR TITLE
[TBOT-587]: Rewrite effects to typescript

### PR DESCRIPTION
**Pull request by aidera**
https://ghostdrive.atlassian.net/browse/TBOT-587

### CODE REVIEW

File: frontend/src/effects/getOneTimeToken.ts [hunk 1]
```diff
@@ -5,12 +5,20 @@ import { AxiosResponse } from 'axios';
...
   filesize?: string;
   filename?: string;
+  isPublic: boolean;
 }
 
 interface TokenResponse {
   jwt_ott: string[];
   user_token: { token: string }[];
-  gateway: { url: string };
+  gateway: {
```
Comment (fifa-master) [inline frontend/src/effects/getOneTimeToken.ts:14]: xaltura

File: frontend/src/effects/file/getFileCid.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import axiosInstance from '../axiosInstance';
+
+interface GetFileCidsResponse {
+  cids: string[];
+  level: string;
+  slug: string;
+  upload_chunk_size: number | null;
+}
+
+export const getFileCids = async ({ slug }: { slug: string }) => {
```
Comment (fifa-master) [inline frontend/src/effects/file/getFileCid.ts:11]: create type

File: frontend/src/effects/balanceEffect.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import axiosInstance from './axiosInstance';
+import { Task } from './types/tasks';
+
+interface History {
```
Comment (fifa-master) [inline frontend/src/effects/balanceEffect.ts:4]: xaltura

File: frontend/src/effects/types/notifications.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export interface Notification {
+  text: string;
+  viewed: boolean;
+  recipient: {
+    id: number;
+  };
+  sender: {
```
Comment (fifa-master) [inline frontend/src/effects/types/notifications.ts:7]: xaltyra

File: frontend/src/effects/EarnEffect.ts [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
+}
+
+type ErrorResponse = AxiosError<{ errors: any }>;
 
 export const checkAllEarnTasks = async () => {
   const url = `${API_PATH}/user/earn`;
 
   try {
-    const { data } = await axiosInstance.get(url);
+    const { data } = await axiosInstance.get<{ data: Task[] }>(url);
```
Comment (fifa-master) [inline frontend/src/effects/EarnEffect.ts:19]: make a generic for such cases```{ data: Task[] }```

File: frontend/src/effects/authorizeUser.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  refresh_token_ttl: number;
+}
+
+export const applyCouponEffect = async (token: string, coupon?: string) => {
+  axiosInstance
+    .post(
+      API_COUPON,
+      { coupon },
+      {
+        headers: {
```
Comment (fifa-master) [inline frontend/src/effects/authorizeUser.ts:36]: does we still need this?

File: frontend/src/effects/authorizeUser.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      }
+    )
+    .catch((err) => console.error(err));
+};
+
+export const authorizeUser = async (reqBody: UserData, ref?: string) => {
+  const res = await axios
+    .post<AuthorizeduserResponse>(API_AUTHORIZATION, reqBody)
+    .then(async (response) => {
+      await setToken(response.data.token);
```
Comment (fifa-master) [inline frontend/src/effects/authorizeUser.ts:48]: async
then
async

File: frontend/src/effects/filesEffects.ts [hunk 8]
```diff
@@ -1,25 +1,27 @@
...
 import { API_PATH } from '../utils/api-urls';
 import axiosInstance from './axiosInstance';
 import { FILE_ACTIONS } from '../config/contracts';
 import { sendFileViewStatistic } from './file/statisticEfect';
 import { getFileCids } from './file/getFileCid';
+import { File, FileDetails, FileToken, PPVFile } from './types/files';
+import { DefaultResponse } from './types/defaults';
 
-export const getDownloadOTT = (body) => {
+export const getDownloadOTT = (body: { slug: string }[]) => {
```
Comment (fifa-master) [inline frontend/src/effects/filesEffects.ts:12]: to new type

File: frontend/src/effects/authorizeUser.ts (hunk 6 above)
Comment (aidera) [reply frontend/src/effects/authorizeUser.ts:36]: hai bude

File: frontend/src/effects/EarnEffect.ts [hunk 9]
```diff
@@ -1,12 +1,22 @@
...
 import { formatPartnerResponce } from '../pages/earn/Partners/utils';
+import { Task } from './types/tasks';
+import { Partner } from './types/partners';
+import { DataWrappedResponse, DefaultResponse } from './types/defaults';
+
+interface CheckTask2Response {
+  success: string;
+}
+
+type ErrorResponse = AxiosError<{ errors: any }>;
```
Comment (fifa-master) [inline frontend/src/effects/EarnEffect.ts:13]: create a generic and move to defaults
this type will be reusable

File: frontend/src/effects/EarnEffect.ts [hunk 10]
```diff
@@ -32,125 +42,136 @@ export const checkTgChatJoin = async () => {
...
 
   try {
-    const { data } = await axiosInstance.post(url, { type: id });
+    const { data } = await axiosInstance.post<DefaultResponse>(url, {
+      type: id
+    });
     if (data?.message === 'success') {
       return 'success';
     } else {
       throw Error();
```
Comment (fifa-master) [inline frontend/src/effects/EarnEffect.ts:171]: Does it not break a code?

File: frontend/src/effects/filesEffects.ts [hunk 11]
```diff
@@ -122,10 +121,10 @@ export const downloadFileEffect = async (file, afterCb) => {
   }
 };
 
-export const autoCompleteSearchEffect = async (term = '') => {
+export const autoCompleteSearchEffect = async (term: string = '') => {
   const url = `${API_PATH}/search/autocomplete?term=${term}`;
   return await axiosInstance
-    .get(url)
+    .get<File[]>(url)
     .then(({ data }) => {
```
Comment (fifa-master) [inline frontend/src/effects/filesEffects.ts:128]: async
then
await

File: frontend/src/effects/filesEffects.ts [hunk 12]
```diff
@@ -169,145 +156,164 @@ export const updateShareEffect = async (
...
+      params: { order_by: 'createdAt', order, page }
     })
-    .catch((err) => {
-      return err;
-    });
-
-export const getFilesEffect = async (page = 1, order = 'desc') => {
-  const url = `${API_PATH}/files?page=${page}&order_by=createdAt&order=${order}`;
-  return await axiosInstance.get(url).then((result) => result.data);
+    .then((result) => result.data);
```
Comment (fifa-master) [inline frontend/src/effects/filesEffects.ts:188]: same 
async
then
await

File: frontend/src/utils/createStarInvoice.ts [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+  title: string;
+  description: string;
+  payload: string;
+  prices: {
+    label: string;
+    amount: number;
+  }[];
+}
+
+const INVOICE_TYPE = {
```
Comment (fifa-master) [inline frontend/src/utils/createStarInvoice.ts:19]: maybe enum for easy use?

File: frontend/src/effects/uploadFileEffect.ts [hunk 14]
```diff
@@ -37,7 +36,7 @@ export const uploadFileEffect = async ({
           data: { user_token, gateway, jwt_ott }
         } = await getOneTimeToken([
           {
-            filesize: file.size,
+            filesize: file.size + '',
```
Comment (fifa-master) [inline frontend/src/effects/uploadFileEffect.ts:39]: do we need this?

File: frontend/src/effects/uploadFileEffect.ts [hunk 15]
```diff
@@ -5,30 +5,29 @@ import {
...
+import { AxiosError } from 'axios';
 
 export const uploadFileEffect = async ({
   files,
   dispatch,
   afterFileUploadCallback,
-  onUploadProgress,
   onUploadError
+}: {
+  files: FileList;
```
Comment (fifa-master) [inline frontend/src/effects/uploadFileEffect.ts:22]: separate to type

File: frontend/src/effects/storageEffects.ts [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+    multiplicator: number;
+    stars: number;
+    storage: number;
+    ton_price: number;
+  } | null;
+}
+
+export const storageListEffect = async () => {
+  return axiosInstance
+    .get<StorageListResponse>(`${API_PATH}/storage/ton`)
```
Comment (fifa-master) [inline frontend/src/effects/storageEffects.ts:19]: async then in whole file 
fix all

**Review by fifa-master — APPROVED** [review]

**Pull request merged**